pkg/minikube/driver/driver.go

```diff
@@ -321,12 +321,16 @@ func SetLibvirtURI(v string) {
 
 // IndexFromMachineName returns the order of the container based on it is name
 func IndexFromMachineName(machineName string) int {
-	// minikube-m02
+	// minikube or offline-docker-20210314040449-6655 or minion-m02
 	sp := strings.Split(machineName, "-")
-	m := strings.Trim(sp[len(sp)-1], "m") // m02
-	i, err := strconv.Atoi(m)
-	if err != nil {
-		return 1
+	m := sp[len(sp)-1]             // minikube or 6655 or m02
+	if strings.HasPrefix(m, "m") { // likely minion node
+		m = strings.TrimPrefix(m, "m")
+		i, err := strconv.Atoi(m)
+		if err != nil {
+			return 1 // master node
+		}
+		return i // minion node
 	}
-	return i
+	return 1 // master node
```

Comment:
do u mind adding unit tests for this ? it is not clear what issue this is fixing and I rather we have clear idea with uni tests

Reply:
do we hve an example failure we could see ?

Reply:
yes, i've given examples in the issue description itself: https://github.com/kubernetes/minikube/issues/10820 and the pr also includes additional tests that should cover those

examples given in the issue were:
```
192.168.79.1 gateway would result in 192.168.79.0 node ip address for name offline-docker-20210314040449-6655
similarly, for node name offline-docker-20210314040449-6654, it would result in 192.168.79.255 node ip
```

Reply:
thanks for the update, would u plz add tests to TestIndexFromMachineName too ? in addition to TestIndexFromMachineNameClusterConfig ?

https://github.com/kubernetes/minikube/blob/1f75634357d1ee7ae44ed02705557353401dd331/pkg/minikube/driver/driver_test.go#L211

Reply:
np, i've added also TestIndexFromMachineName tests